### PR TITLE
Switch to UsePathBase instead of relying on Kestrel feature #236

### PR DESCRIPTION
**Pull request by JunTaoLuo**
#236

### CODE REVIEW

File: test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs [hunk 1]
```diff
@@ -79,14 +78,64 @@ public void UrlDelayRegistered()
...
+                    app.Run(context => Task.FromResult(0));
+                });
 
             Assert.Null(builder.GetSetting(WebHostDefaults.ServerUrlsKey));
 
             // Adds a server and calls Build()
             var server = new TestServer(builder);
 
-            Assert.Equal("http://localhost:12345/", builder.GetSetting(WebHostDefaults.ServerUrlsKey));
+            Assert.Equal("http://localhost:12345", builder.GetSetting(WebHostDefaults.ServerUrlsKey));
```
Comment (JunTaoLuo) [inline test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs]: Slight behaviour change. Paths that represent empty paths will no longer be appended to the url address that the server sees.

File: test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs [hunk 2]
```diff
@@ -79,14 +78,64 @@ public void UrlDelayRegistered()
                 .UseSetting("TOKEN", "TestToken")
                 .UseSetting("PORT", "12345")
                 .UseSetting("APPL_PATH", "/")
-                .UseIISIntegration();
+                .UseIISIntegration()
+                .Configure(app =>
+                {
+                    app.Run(context => Task.FromResult(0));
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs]: why did you need to add this?

**Comment by Tratcher** [comment]
:shipit:

File: test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs (hunk 2 above)
Comment (JunTaoLuo) [inline test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs]: Aesthetics. Otherwise you get this in the test output:
```
Application startup exception: System.InvalidOperationException: No service for type 'Microsoft.AspNetCore.Hosting.IStartup' has been registered.
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.EnsureStartup()
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.EnsureApplicationServices()
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.BuildApplication()
```

File: src/Microsoft.AspNetCore.Server.IISIntegration/IISSetupFilter.cs [hunk 3]
```diff
@@ -4,22 +4,26 @@
...
         {
             _pairingToken = pairingToken;
+            _pathBase = pathBase;
         }
 
         public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next)
         {
             return app =>
             {
+                app.UsePathBase(_pathBase);
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.IISIntegration/IISSetupFilter.cs]: Can we just avoid the middleware if the pathbase is empty?

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.IISIntegration/IISSetupFilter.cs]: UsePathBase already does that

Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Server.IISIntegration/IISSetupFilter.cs]: That's being done in the middleware.
